pkg/eventfilter/subscriptionsapi/exact_filter.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subscriptionsapi
+
+import (
+	"context"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"go.uber.org/zap"
+	"knative.dev/pkg/logging"
+
+	"knative.dev/eventing/pkg/eventfilter/attributes"
+
+	"knative.dev/eventing/pkg/eventfilter"
+)
+
+type exactFilter struct {
+	attribute, value string
+	attrsFilter      eventfilter.Filter
+}
+
+// NewExactFilter returns an event filter which passes if value exactly matches the value of the context
+// attribute in the CloudEvent.
+func NewExactFilter(attribute, value string) eventfilter.Filter {
```

Comment:
From the spec for `exact`, `prefix` and `suffix`:
```
The attribute name and value specified in the filter express MUST NOT be empty strings.
```

Shouldn't the constructor return an error as the second parameter just as the `sql` does?